spec/attachment2xml.spec.js

```diff
@@ -0,0 +1,34 @@
+/* eslint-env node, jasmine */
+'use strict';
+
+const attachmentToJson = require('../lib/actions/attachmentToJson');
+const json = require('./data/po.json');
+const { expect } = require('chai');
+const nock = require('nock');
+const streamString = require('stream-string');
+
+
+require('chai').should(); // expect is assertion styles used my elastic.io
+
+describe('should convert XML attachment 2 JSON', () => {
+    const mockSever = 'http://test.env.mock';
+
+    before(function testInit() {
+        nock(mockSever)
+          .get('/')
+          .replyWithFile(200, 'spec/data/po.xml');
+    });
+
+    it('Convert attachment to json', async () => {
+        const msg = {
+            attachments: {
+                'po.xml': {
+                    url: mockSever
+                }
+            }
+        };
+        const { body } = await attachmentToJson.process(msg, {});
```

Comment:
In theory this should work, but I recall that there were some bugs that I could not pin down related to returning messages instead of emitting them on the platform.

Reply:
ok this is what used in the other test case